proto/groups/v1alpha/groups.proto

```diff
@@ -0,0 +1,204 @@
+
+  // Lists content of a group.
+  rpc ListContent(ListContentRequest) returns (ListContentResponse);
+
+  // Lists groups.
+  rpc ListGroups(ListGroupsRequest) returns (ListGroupsResponse);
+}
+
+// Request to create a group.
+message CreateGroupRequest {
+  // Required. Title of the new group.
+  string title = 1;
+
+  // Optional. Description of the new group.
+  string description = 2;
+
+  // Optional. List of initial members for the new group.
+  // Members can also be managed with separate requests after group is already created.
+  map<string, Role> members = 3;
+}
+
+// Request to get a group.
+message GetGroupRequest {
+  // Required. ID of the group to get.
+  // Must be a fully-qualified EID.
+  string id = 1;
+
+  // Optional. Version of the group to get information about.
+  // If empty, latest_version is will be used.
+  string version = 2;
+}
+
+// Request to update a group.
+// Some fields are required, even if unchanged,
+// to distinguish between setting the value to an empty string, from an unchanged value.
+// Read the comments!
+message UpdateGroupRequest {
+  // ID of the group to update.
+  string id = 1;
+
+  // Required. Title of the Group.
+  // Must always be provided, even if unchanged.
+  string title = 2;
+
+  // Required. Description of the Group. Can be empty string.
+  // Must always be provided, even if unchanged.
+  string description = 3;
+
+  // Optional. List of members to be updated in the Group.
+  // Key is the member's Account ID,
+  // value is the Role.
+  // To remove a member from the group, set the role to unspecified.
+  map<string, Role> updated_members = 4;
+
+  // Optional. List of content to be updated in the Group.
+  // Key is a pretty path on which the content is published,
+  // value is a Hyperdocs URL of the content.
+  // To unpublish content set the pretty path to an empty string.
+  map<string, string> updated_content = 5;
+}
+
+// Response to list members.
+message ListMembersRequest {
+  // Required. ID of the group to list members of.
+  string id = 1;
+
+  // Optional. Version of the group to list members of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
+
+  // Optional. Maximum number of members to return.
+  int32 page_size = 3;
+
+  // Optional. Page token to continue listing members from.
+  string page_token = 4;
+}
+
+// Response to list members.
+message ListMembersResponse {
+  // ID of the group owner.
+  string owner_account_id = 1;
+
+  // List of members of the group.
+  map<string, Role> members = 2;
+
+  // Token to continue listing members from.
+  string next_page_token = 3;
+}
+
+// Request to list content.
+message ListContentRequest {
+  // Required. ID of the group to list content of.
+  string id = 1;
+
+  // Optional. Version of the group to list content of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
```

Comment:
According to its members? I don't understand, I thought it'd be according to the local state of the group.

Reply:
According to the information from group members that we have locally. It's the same definition as the field `versión` on the Group type.

The thing is, with groups we don't care about trusted peers or untrusted peers. We only really care about how members and owner of the group are seeing it, so that's the version we use by default.